src/core/effects/qgsshadoweffect.cpp

```diff
@@ -174,8 +181,18 @@ void QgsShadowEffect::readProperties( const QVariantMap &props )
 QRectF QgsShadowEffect::boundingRect( const QRectF &rect, const QgsRenderContext &context ) const
 {
   //blur radius and offset distance
-  const int blurLevel = std::round( context.convertToPainterUnits( mBlurLevel, mBlurUnit, mBlurMapUnitScale ) );
+  int blurLevel = std::round( context.convertToPainterUnits( mBlurLevel, mBlurUnit, mBlurMapUnitScale ) );
+
   double spread = context.convertToPainterUnits( mOffsetDist, mOffsetUnit, mOffsetMapUnitScale );
+
+  if ( context.flags() & QgsRenderContext::Flag::RenderSymbolPreview )
+  {
+    // avoid excessively large blur or offset in symbol preview icons -- it's too slow to calculate, and unnecessary
+    // for just a preview icon
+    blurLevel = std::min( blurLevel, 30 );
+    spread = std::min( spread, 100.0 );
```

Comment:
Is that a reason why in GlowEffect the minimum spread is set to 50 and here to 100?
Should such values be put in constants to avoid such things and make them easier to change if needed?

Reply:
Ah, it's a badly named variable. It's actually "offset" for the shadow effect. I'll fix.

Reply:
I've reworked this now to avoid the duplicate values